jtsync/jtsync.rb

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env ruby
+#
+# Copyright 2016, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'trello'
+require 'netrc'
+require 'optparse'
+require 'ostruct'
+
+TRELLO_BOARD = "ywSqlQpZ".freeze
+
+# Map all jenkins job names to trello list/card combinations
+MAPPING = {
+  suse: {
+    matrix: lambda do |job|
+      version = job.project.split(':')[2]
+      ["Cloud #{version}", "C#{version} #{job.name}"]
+    end,
+    normal: lambda do |job|
+      version = job.name.split('-')[1]
+      ["Cloud #{version[-1, 1]}", job.name]
+    end
+  },
+  opensuse: {
+    matrix: lambda do |job|
+      name = job.name.gsub('openstack-', '')
+      ['OpenStack', "#{name}: #{job.project}"]
+    end,
+    normal: lambda do |job|
+      ['OpenStack', job.name]
+    end
+  }
+}.freeze
+
+STATUS_MAPPING = {
+  '0' => 'successful',
+  '1' => 'failed'
+}.freeze
+
+class Job < Struct.new(:type, :service, :project, :name, :status)
+end
+
+def credentials_from_netrc
+  netrc = Netrc.read
+  dev, member = netrc['trello.com']
```

Comment:
Should be `api.trello.com`, as API requests go there. Once a lib natively supports netrc then we would have to provide the credentials for  `api.trello.com` anyway.